### PR TITLE
Create Disk and enable protection with Azure Backup

### PR DESCRIPTION
**Pull request by nilaydshah**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Template creates following resources: Create disk, Create backup vault and policy, Create role assignment, Configure protection

### CODE REVIEW

File: 101-backup-create-disk-enable-protection/metadata.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "$schema": "https://aka.ms/azure-quickstart-templates-metadata-schema#",
+  "type": "QuickStart",
+  "itemDisplayName": "Create Disk & enable protection via Backup Vault",
+  "description": "Template that creates disk and enable protection via Backup Vault",
```
Comment (bmoore-msft) [inline 101-backup-create-disk-enable-protection/metadata.json:5]: Template that creates a disk and enables protection via a Backup Vault

File: 101-backup-create-disk-enable-protection/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+        "creationData": {
+          "createOption": "Empty"
+        },
+        "diskSizeGB": 200
+      }
+    },
+    {
+      "type": "Microsoft.Authorization/roleAssignments",
+      "apiVersion": "2020-04-01-preview",
+      "name": "[parameters('roleNameGuidForDisk')]",
```
Comment (bmoore-msft) [inline 101-backup-create-disk-enable-protection/azuredeploy.json:176]: this must be deterministic and unique for the perms, role and scope (the parameter is not)... e.g.

```guid(resourceGroup().id, variables('roleDefinitionIdForDisk'), resourceId('Microsoft.DataProtection/backupVaults', parameters('vaultName'))```

File: 101-backup-create-disk-enable-protection/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+      ],
+      "properties": {
+        "roleDefinitionId": "[variables('roleDefinitionIdForDisk')]",
+        "principalId": "[reference(resourceId('Microsoft.DataProtection/backupVaults/', parameters('vaultName')), '2020-01-01-alpha', 'Full').identity.principalId]"
+      }
+    },
+    {
+      "type": "Microsoft.Authorization/roleAssignments",
+      "apiVersion": "2020-04-01-preview",
+      "name": "[parameters('roleNameGuidForSnapshotRG')]",
```
Comment (bmoore-msft) [inline 101-backup-create-disk-enable-protection/azuredeploy.json:190]: same as the other RA

**Review by bmoore-msft — CHANGES_REQUESTED** [review]
See comments, also - we're no longer accepting samples in the root of the repo, see the contribution guide - this will need to be moved to the appropriate folder under the ```quickstarts``` folder

File: 101-backup-create-disk-enable-protection/azuredeploy.json (hunk 3 above)
Comment (nilaydshah) [reply 101-backup-create-disk-enable-protection/azuredeploy.json]: Addressed this.

File: 101-backup-create-disk-enable-protection/azuredeploy.json (hunk 2 above)
Comment (nilaydshah) [reply 101-backup-create-disk-enable-protection/azuredeploy.json]: Addressed this.

File: 101-backup-create-disk-enable-protection/metadata.json (hunk 1 above)
Comment (nilaydshah) [reply 101-backup-create-disk-enable-protection/metadata.json]: Addressed this.

**Review by nilaydshah — COMMENTED** [review]
Addressed the comments. Moved the template from root to appropriate subfolder.

**Comment by nilaydshah** [comment]
> See comments, also - we're no longer accepting samples in the root of the repo, see the contribution guide - this will need to be moved to the appropriate folder under the `quickstarts` folder

Addressed this.

**Pull request merged**